### DISCUSSION TITLE
Remove naming restrictions from Index UID

### DISCUSSION THREAD

**Discussion by sshankar-rks** [meilisearch/product#571] [Feedback & Feature Proposal]
**Description**
As per documentation, an index does not have a name, only a UID. Trying to create an Index with name "A.B" results in an error message: "message":"invalid index uid `A.B`, the uid must be an integer or a string containing only alphanumeric characters a-z A-Z 0-9, hyphens - and underscores _."

I want to use entity scoped names for indexes. There can be more than one index into a entity. Having scoped names helps me quickly see what entity and index I'm dealing with. Underscores can be used for the same purpose, but are less readable and ambiguous (have to use underscores for spaces as well).

**Expected behavior**
Either allow index to have a name (and no naming restrictions on that field).
(OR)
Remove restrictions from uid

**Meilisearch version:** 
v0.29.0

**Comment by oluademola** [comment]
Hello @sshankar-rks 

This is indeed the expected behavior of the search engine, as indicated in the docs and the error message. I believe this is not a bug but a feature request. We will move this to the appropriate repo.

We appreciate your feedback. Thank you very much.
